protocol/communities/communnity_privileged_member_sync_msg.go

```diff
@@ -88,3 +90,56 @@ func (m *Manager) HandleSyncAllRequestToJoinForNewPrivilegedMember(message *prot
 	}
 	return nonAcceptedRequestsToJoin, nil
 }
+
+func (m *Manager) HandleEditSharedAddressesPrivilegedUserSyncMessage(message *protobuf.CommunityPrivilegedUserSyncMessage, community *Community) error {
+	if !(community.IsTokenMaster() || community.IsOwner()) {
+		return ErrNotEnoughPermissions
+	}
+
+	publicKey := message.SyncEditSharedAddresses.PublicKey
+	editSharedAddress := message.SyncEditSharedAddresses.EditSharedAddress
+	if err := community.ValidateEditSharedAddresses(publicKey, editSharedAddress); err != nil {
+		return err
+	}
+
+	return m.handleCommunityEditSharedAddresses(publicKey, community.ID(), editSharedAddress.RevealedAccounts, message.Clock)
+}
+
+func (m *Manager) processPrivilegedUserSharedRequestToJoin(community *Community, requestToJoin *RequestToJoin) error {
+	existingRequestToJoin, err := m.persistence.GetRequestToJoin(requestToJoin.ID)
+	if err != nil && err != sql.ErrNoRows {
+		return err
+	}
+
+	member, exists := community.Members()[requestToJoin.PublicKey]
+	if exists && existingRequestToJoin != nil && (existingRequestToJoin.Clock > requestToJoin.Clock || member.LastUpdateClock > requestToJoin.Clock) {
```

Comment:
Wouldn't `existingRequestToJoin` be `nil` if we have `ErrNoRows` above? If so, we probably have to guard against it here to avoid a crash